### ISSUE TITLE
Bug: selector-no-qualifying-type false positive with nested SCSS and BEM naming

### ISSUE THREAD

**Issue by inoplanetyanya** [stylelint/stylelint#8978]
### What minimal example or steps are needed to reproduce the bug?

test.scss:
```
.cost {
  &__item {
    display: flex;

    .options {
      display: flex;
    }
  }
}
```

### What minimal configuration is needed to reproduce the bug?

```
/** @type {import('stylelint').Config} */
export default {
  extends: ["stylelint-config-standard-scss"],
  rules: {
    "selector-no-qualifying-type": true
  }
};
```

### How did you run Stylelint?

```
npx stylelint "test.scss"
```

### Which Stylelint-related dependencies are you using?

```
{
  "devDependencies": {
    "sass": "^1.97.2",
    "stylelint": "^17.0.0",
    "stylelint-config-standard-scss": "^17.0.0",
    "stylelint-scss": "^7.0.0"
  }
}
```

### What did you expect to happen?

no errors

### What actually happened?

```
test.scss
  5:5  ✖  Unexpected qualifying type selector "__item" in "__item.cost"  selector-no-qualifying-type

✖ 1 problem (1 error, 0 warnings)
```

### Do you have a proposal to fix the bug?

nope

**Comment by jeddy3** [comment]
@inoplanetyanya Thanks for the report and for using the template.

The `selector-no-qualifying-type` rule was changed in 17.0.0 to use standard CSS nesting (see the [migration guide](https://stylelint.io/migration-guide/to-17#changed-no-duplicate-selectors-and-selector-no-qualifying-type-for-standard-css-nesting)). It is incompatible with the SCSS nesting that concatenates strings.

You shouldn't turn the rule on for SCSS (see [the rule docs](https://stylelint.io/user-guide/rules/selector-no-qualifying-type/)).

You've correctly opened an issue (https://github.com/stylelint-scss/stylelint-scss/issues/1230) in the stylelint-scss issue tracker. The SCSS community may want to consider creating their own custom version of the rule that uses SCSS nesting.
